common/persistence/sql/sqlTaskManager.go

```diff
@@ -282,8 +298,14 @@ func (m *sqlTaskManager) ListTaskQueue(request *persistence.ListTaskQueueRequest
 			Name:        lastRow.Name,
 			TaskType:    lastRow.TaskType,
 		})
-	case pageToken.ShardID+1 < m.nShards:
-		nextPageToken, err = gobSerialize(&taskQueuePageToken{ShardID: pageToken.ShardID + 1, TaskType: math.MinInt16})
+	case shardLessThan < math.MaxUint32:
+		var lastShardId uint32
+		if len(rows) == 0 {
+			lastShardId = shardLessThan
+		} else {
+			lastShardId = rows[len(rows)-1].ShardID
+		}
+		nextPageToken, err = gobSerialize(&taskQueuePageToken{ShardID: lastShardId+1, TaskType: math.MinInt16})
```

Comment:
`ShardID: lastShardId+1` can this result into losing taskQueues if pageSize restricts all the taskQueues to be returned back with that ShardID.